app/src/main/java/com/mapbox/android/gestures/testapp/MainActivity.java

```diff
@@ -0,0 +1,317 @@
+      @Override
+      public boolean onScroll(MotionEvent e1, MotionEvent e2, float distanceX, float distanceY) {
+        icon.setTranslationX(icon.getTranslationX() - distanceX);
+        icon.setTranslationY(icon.getTranslationY() - distanceY);
+        return true;
+      }
+
+      @Override
+      public boolean onDoubleTap(MotionEvent e) {
+        rescaleIcon(1.40f);
+        return true;
+      }
+    });
+
+    androidGesturesManager.setMultiFingerTapGestureListener(
+      new MultiFingerTapGestureDetector.OnMultiFingerTapGestureListener() {
+
+        @Override
+        public boolean onMultiFingerTap(MultiFingerTapGestureDetector detector, int pointersCount) {
+          if (pointersCount == 2) {
+            rescaleIcon(0.65f);
+          }
+          return true;
+        }
+      });
+
+    androidGesturesManager.setShoveGestureListener(new ShoveGestureDetector.SimpleOnShoveGestureListener() {
+      @Override
+      public boolean onShove(ShoveGestureDetector detector, float deltaPixelsSinceLast, float deltaPixelsSinceStart) {
+        icon.setRotationX(icon.getRotationX() - deltaPixelsSinceLast);
+        return true;
+      }
+    });
+  }
+
+  @Override
+  public boolean onTouchEvent(MotionEvent event) {
+    return androidGesturesManager.onTouchEvent(event) || super.onTouchEvent(event);
+  }
+
+  @Override
+  public boolean onCreateOptionsMenu(Menu menu) {
+    MenuInflater inflater = getMenuInflater();
+    inflater.inflate(R.menu.menu, menu);
+    return true;
+  }
+
+  @Override
+  public boolean onOptionsItemSelected(MenuItem item) {
+    if (item.getItemId() == R.id.menu_help) {
+      HelpDialogFragment helpDialogFragment =
+        (HelpDialogFragment) getFragmentManager().findFragmentByTag(HelpDialogFragment.TAG);
+
+      if (helpDialogFragment == null) {
+        helpDialogFragment = HelpDialogFragment.newInstance();
+        getFragmentManager()
+          .beginTransaction()
+          .add(helpDialogFragment, HelpDialogFragment.TAG)
+          .commit();
+      }
+      return true;
+    }
+    return false;
+  }
+
+  private void rescaleIcon(float scaleFactor) {
+    DisplayMetrics metrics = new DisplayMetrics();
+    getWindowManager().getDefaultDisplay().getMetrics(metrics);
+
+    ViewGroup.LayoutParams layoutParams = icon.getLayoutParams();
+    int width = (int) (layoutParams.width * scaleFactor);
+    int height = (int) (layoutParams.height * scaleFactor);
+    if (width > 100 && height > 100 && width < metrics.widthPixels && height < metrics.heightPixels) {
+      layoutParams.width = width;
+      layoutParams.height = height;
+      icon.setLayoutParams(layoutParams);
+    }
+  }
+
+  private void setVelocityEnabled(boolean enabled) {
+    if (enabled) {
+      velocityEnabled = true;
+      velocityButton.setText(R.string.velocity_on);
+    } else {
+      velocityEnabled = false;
+      velocityButton.setText(R.string.velocity_off);
+    }
+  }
+
+  private class ExclusiveObject {
```

Comment:
Don't really understand the concept of the class from the name: it excludes objects?

Reply:
The purpose of the class is to provide the `String` representation of the gestures for the `Spinner`. I'll try with `ExclusiveSetSpinnerObject` to make it clearer.